keyboards/ergodox_stm32/rules.mk

```
@@ -0,0 +1,30 @@
+SRC += matrix.c \
+drivers/arm/i2c_master.c
```

You don't need the full path here, actually.  Specifically, the drivers folder is added to the VPATH (virtual path) for the project, so it should pick it up, without the full path. 
```suggestion
    i2c_master.c
```

---

And actually.... this should be added seperately:

```c
QUANTUM_LIB_SRC += i2c_master.c
```